### ISSUE TITLE
Got error when trying to connect to Redis as the chart repository cache in Harbor chart

### ISSUE THREAD

**Issue by ywk253100** [goharbor/harbor#6046]
When the core trying to connect to the Redis while the Redis is not ready yet, got error:
```
2018-10-15T07:44:44Z [ERROR] [cache.go:168]: Failed to initialize redis cache: dial tcp 10.97.140.241:6379: getsockopt: connection timed out
2018-10-15T07:44:44Z [INFO] No cache is enabled for chart caching
```

We should considering to set a shorter timeout and retry n times before reporting the failure.

**Comment by steven-zou** [comment]
Move to SP47 as no effort to complete it in sp46

**Comment by steven-zou** [comment]
Code merged in #6379. Close issue.
